files/base.rb

```diff
@@ -169,6 +188,18 @@ def full_description_hash
       'Tip' => tip,
       'Server' => Socket.gethostname,
     }
+
+    if event_description != 'N/A'
+        description_hash['description'] = event_description
+    end
+
+    if component != 'N/A'
+      if ! component.empty?
+        description_hash['component'] = '[' + component.join(',') + ']'
```

Comment:
We don't need this logic right?

Reply:
Yes. I am already doing this in the getter. I will change this.

The only question I have regarding this is if you want this to be passed on straight to the hash. I think you wanted these fields as optional . However, if there is no issue in them appearing N/A downstream then I will go ahead and add it straight to the hash

Reply:
I don't have an issue with having them appear as N/A downstream.